src/makeInfinityAdder.js

```diff
@@ -6,7 +6,21 @@
  */
 
 function makeInfinityAdder() {
-  // write code here
+  let sum = 0;
+
+  return function add(a) {
+    if (a === undefined) {
+      const answer = sum;
+
+      sum = 0;
+
+      return answer;
+    } else {
+      sum += a;
+
+      return add;
+    }
```

Comment:
```suggestion
      return answer;
    }
    
    sum += a;

    return add;
```
No need for else. You code will not go further after return anyway